instantsearch-utils/build.gradle.kts

```diff
@@ -1,15 +1,69 @@
 plugins {
     kotlin("multiplatform")
+    id("com.android.library")
     id("com.vanniktech.maven.publish")
 }
 
+android {
+    namespace = "com.algolia.instantsearch.utils"
+    compileSdk = 33
+
+    defaultConfig {
+        minSdk = 21
+        testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+    }
+
+    compileOptions {
+        sourceCompatibility = JavaVersion.VERSION_1_8
+        targetCompatibility = JavaVersion.VERSION_1_8
+    }
+
+    buildFeatures {
+        buildConfig = false
+    }
+
+    testOptions.unitTests.apply {
+        isIncludeAndroidResources = true
+        isReturnDefaultValues = true
+    }
+}
+
 kotlin {
     explicitApi()
     jvm()
+    android {
+        publishAllLibraryVariants()
+        publishLibraryVariantsGroupedByFlavor = true
+    }
 
     sourceSets {
         all {
-            languageSettings.optIn("kotlin.RequiresOptIn")
+            languageSettings.optIn("com.algolia.instantsearch.InternalInstantSearch")
```

Comment:
Could you explain briefly what does this line do?

Reply:
`@InternalInstantSearch` is an annotation to mark APIs as internal to InstantSearch (i.e. not meant to be used by the library users), and produce a warning message if a marked API is used. The warning can be suppressed with either `@OptIn` annotation or `optIn` setting in gradle config.

The change above suppresses the usage warnings in the library itself since it's meant to be used inside it.

Reply:
Thanks. Otherwise, everything is good 👍